internal/database/badger/race_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package badger
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Database_threadSafety(t *testing.T) {
+	// This test consists in checking for concurrent access
+	// using the -race detector.
+	t.Parallel()
+
+	var startWg, endWg sync.WaitGroup
```

Comment:
I think we should be able to use just one wait group. Let me know what I might be missing.

Reply:
We want to wait for all goroutines (aka workers) to be ready to make sure they all operate at the same time (since starting a goroutine takes a few milliseconds) using `startWg`, and then we want to wait for all of them to exit (`endWg`)